Dockerfile.mri.erb

```diff
@@ -148,7 +149,7 @@ RUN bash -c " \
 RUN find /usr/local/rake-compiler/ruby/*linux*/ -name libruby.so | xargs rm
 RUN find /usr/local/rake-compiler/ruby/*linux*/ -name libruby-static.a | while read f ; do cp $f `echo $f | sed s/-static//` ; done
 RUN find /usr/local/rake-compiler/ruby/*linux*/ -name libruby.a | while read f ; do ar t $f | xargs ar d $f ; done
-RUN find /usr/local/rake-compiler/ruby/*linux*/ -name mkmf.rb | while read f ; do sed -i ':a;N;$!ba;s/TRY_LINK = [^\n]*\n[^\n]*\n[^\n]*LOCAL_LIBS)/& -lruby-static -lpthread -lrt -ldl <% if platform =~ /x86_64/ %> -lcrypt <% if platform !~ /musl/ %> -lz <% end %> <% end %>/' $f ; done
```

Comment:
This reverts the fix for https://github.com/rake-compiler/rake-compiler-dock/issues/65, is that intentional (and does it work?)

Reply:
Reading the description of the original bug, this may be what's causing the failure of the x86_64-linux-gnu build?

Reply:
Now it works! I added `ac_cv_*` variables to disable the libcrypt and libz usage when the cross platform matches the native one. And added this to the commit message.

Reply:
I had no idea you could do that! Nice.